app/models/option_set.rb

```diff
@@ -24,56 +24,64 @@ class OptionSet < ActiveRecord::Base
   # the dependent object doesn't know who destroyed it.
   before_destroy { report_option_set_choices.each(&:option_set_destroyed) }
 
-  scope :by_name, -> { order('option_sets.name') }
+  # Generate sequence
+  after_save :generate_sequence
```

Comment:
I'm not sure about this approach because it would potentially re-number existing nodes, correct? That could be an issue if e.g. a few nodes get added to an existing set and the form is already out in the wild.

I think sequence num should be fixed once it is assigned.

Perhaps in OptionSet.update_children you could get the current max sequence num for the option set right before you create each node. It won't be great performancewise but that code needs to go eventually anyway, and it isn't used for large option set imports...